File: hw/collahuazo/u1/hw07Collections/hw07CollectionsPython/view/CollectionControlerApp.py
import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from datetime import date
from controller.Person import Person

def print_the_collection(things):
    print(f"\nsize of things --> {len(things)}")
    print(f"These are my things --> \n{things}")

def main():
    # Primera persona
    id_val = 1
    full_name = "Brandon Collahuazo"
    born_on_date = date(1999, 3, 4)
    alive = True
    person = Person(id_val, full_name, born_on_date, alive)

    print(person)
    print(f"Age of person 1 is {person.compute_age_in_years()}")

    # Segunda persona
    id_val = 2
    full_name = "Elizabeth Cumbajin"
    born_on_date = date(1962, 4, 23)
    alive = True
    person2 = Person(id_val, full_name, born_on_date, alive)

    # Colección heterogénea (Como tu 'Collection things')
    things = [] # En Python las listas aceptan cualquier tipo de dato
    print(f"size of things --> {things}")

    things.append(8000)
    things.append(3.5)
    things.append("Quito")
    things.append(True)

    print_the_collection(things)

    things.append(4078.76)
    things.append("a")

    print_the_collection(things)

    # Lista de objetos Person (Como tu 'ArrayList<Person> people')
    people = []
    people.append(person)
    people.append(person2)

    print(f"\npeople --> {people}")
    print("")

    # ForEach equivalente
    for p in people:
        print(f"Person --> {p}")

    print("")
    
    # Bucle for para agregar más "pepitos"
    # range(2, 7) genera: 2, 3, 4, 5, 6
    for i in range(2, 7):
        people.append(Person(i + 1, "pepito", date.today(), True))

    # Imprimir la lista final actualizada
    for p in people:
        print(f"Person --> {p}")

if __name__ == "__main__":
    main()

File: ws/collahuazo/u1/ws08Collections/src/ec/edu/espe/view/CollectionControlerApp.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.view;

import ec.edu.espe.model.Person;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Collection;

/**
 *
 * @author Brandon Collahuazo, code Bros, @ESPE
 */
public class CollectionControlerApp {

    public static void main(String[] args) {
        int id;
        String fullName;
        LocalDate bornOnDate;
        boolean alive;

        id = 1;
        fullName = "Brandon Collahuazo";
        bornOnDate = LocalDate.of(1999, 03, 04);
        alive = true;
        Person person = new Person(id, fullName, bornOnDate, alive);

        System.out.println(person);
        System.out.println("Age of person 1 is " + person.computeAgeInYears());

        id = 2;
        fullName = "Elizabeth Cumbajin";
        bornOnDate = LocalDate.of(1962, 04, 23);
        alive = true;
        Person person2 = new Person(id, fullName, bornOnDate, alive);

        Collection things;
        things = new ArrayList<>();
        System.out.println("size of things -->" + things);

        things.add(8000);
        things.add(3.5F);
        things.add("Quito");
        things.add(true);

        printTheCollection(things);

        things.add(4078.76F);
        things.add("a");

        printTheCollection(things);

        ArrayList<Person> people = new ArrayList<>();
        people.add(person);
        people.add(person2);

        System.out.println("\npeople --> " + people);
        System.out.println("");

        people.forEach((p) -> {
            System.out.println("Person --> " + p);
        });
        
        System.out.println("");
        for (int i = 2; i < 7; i++) {
            people.add(new Person(i + 1, "pepito", LocalDate.now(), true));
        }
        people.forEach((p) -> {
            System.out.println("Person --> " + p);
        });
    }

    private static void printTheCollection(Collection things) {
        System.out.println("\nsize of things --> " + things.size());
        System.out.println("These are my things --> \n" + things);
    }

}

